src/main/java/me/rkfg/xmpp/bot/plugins/FaggotOfTheDayPlugin2.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Plugin for Faggot of the Day game in MUCs.
+ * Copyright (C) 2017 Kona-chan
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package me.rkfg.xmpp.bot.plugins;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.muc.MultiUserChat;
+import org.jivesoftware.smackx.muc.Occupant;
+import org.jxmpp.util.XmppStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ru.ppsrk.gwt.client.ClientAuthException;
+import ru.ppsrk.gwt.client.LogicException;
+
+/**
+ * Lets users of MUCs play Faggot of the Day.
+ *
+ * @author Kona-chan
+ * @version 0.3.0
+ */
+public class FaggotOfTheDayPlugin2 extends CommandPlugin {
+
+    private static final List<String> ALL_COMMANDS = Arrays.asList("pidor", "пидор");
+    private static final String INFO_FAGGOT_PENDING = "Пидор дня вычисляется.";
+    private static final String INFO_FAGGOT_IS = "сегодня Пидор дня — ";
+    private static final String INFO_FAGGOT_IS_YOU = INFO_FAGGOT_IS + "ты!";
+
+    private boolean isListening = false;
+    private Timer timer;
+
+    private final Set<Occupant> occupants = new HashSet<>();
+    private Occupant faggot;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final Random random = new Random();
+
+    @Override
+    public String processCommand(Message message, Matcher matcher)
+            throws LogicException, ClientAuthException {
+        startListening();
+        startTimer();
```

Comment:
Works for timer, doesn’t really work for listener since at the time of plugin intialization we haven’t joined any MUC yet.